passive/Find Data Store Credentials.js

```diff
@@ -0,0 +1,98 @@
+// 10-JAN-2016
+// Josh Bowser
+// jkbowser[at]gmail[dot]com
+// 
+// Credits to Mikey and Hillj for the sanity check
+// Credit to http://regexr.com allowing me to test my shakey regexp
+//***************************************
+
+//This JavaScript will tell ZAP to regexp test for some well known 
+//RDBMS connection string key words in a response body to see if any are leaked out to the client
+//Unlikely to find matches?  Yes, but I've seen this done twice now in various pen tests
+//so I figured it was time to add a script to check for it
+
+function scan(ps, msg, src)
+{
+    url = msg.getRequestHeader().getURI().toString(); //For tracking the request
+    body = msg.getResponseBody().toString() //get the body so we can test against our regexp
+    alertRisk = 3 //High risk hardcoded
+    alertReliability = 1 //low confidence hardcoded
+    alertTitle = "Database Credentials Exposed"
+    alertDesc = "Database credentials are exposed to the client. Review Other Info carefully and manually inspect the response to confirm"
+    alertSolution = "Do not pass database connection strings, key/value pair data or credentials to the client"
+    cweId = 0 //?
+    wascId = 0  //?
+
+	//Regexp for different connection string keywords. Tried to group them for readability. 
+
+	//server
+	svr = /server\s*=([^;]*);?/gi //match any case-insensitive to: server<possible space(s)>=<any non-semicolon char>;<optional ;>
+
+	//database names
+	db = /database\s*=([^;]*);?/gi //match any case-insensitive to: database<possible space(s)>=<any non-semicolon char>;<optional ;>
+	ds = /data source\s*=([^;]*);?/gi //match any case-insensitive to: data source<possible space(s)>=<any non-semicolon char>;<optional ;>
+	ic = /initial catalog\s*=([^;]*);?/gi //match any case-insensitive to: initial catalog<possible space(s)>=<any non-semicolon char>;<optional ;>
+	
+	//usernames
+	un = /user\s?name\s*=([^;]*);?/gi //match any case-insensitive to: user<possible space>name<possible space(s)>=<any non-semicolon char>;<optional ;>
+	ui = /user\s?id\s*=([^;]*);?/gi //match any case-insensitive to: user<possible space>id<possible space(s)>=<any non-semicolon char>;<optional ;>
+	uid = /uid\s*=([^;]*);?/gi //match any case-insensitive to: uid<possible space(s)>=<any non-semicolon char>;<optional ;>
+
+	//password
+	pw = /password\s*=([^;]*);?/gi //match any case-insensitive to: password<possible space(s)>=<any non-semicolon char>;<optional ;>
+	pwd = /pwd\s*=([^;]*);?/gi //match any case-insensitive to: pwd<possible space(s)>=<any non-semicolon char>;<optional ;>
+
+	//dump into array
+	dataSourceCheck = [svr, db, ds, ic, un, ui, uid, pw, pwd]
+
+	//create an array for matching strings as we might have more than 1 hit
+	var credsFound = []
+
+	//now for loop through the array
+	for (var i=0; i < dataSourceCheck.length; i++)
+	{
+
+		//If test the body against our regexp array
+		if(dataSourceCheck[i].test(body)) //test() returns true/false boolean so we know to continue
```

Comment:
Is this check (and following statement) really needed? It could just do the exec(s). The code is not harder to read without it (it's thoroughly commented anyway).